### ISSUE TITLE
[Agents][Windows 10]: "Failed to render configuration" error logs when windows agent is installed with only System integration.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/beats#27057]
**Kibana version:** 7.14.0 BC-4 Kibana cloud environment

**Host OS and Browser version:** Windows 10 x64, All

**Build details:**
```
Build: 42656
Commit: 82a4f6a7fa23946667599787cdb1b2d82c1d0dfb
Artifact Link: https://staging.elastic.co/7.14.0-b3779639/downloads/beats/elastic-agent/elastic-agent-7.14.0-windows-x86_64.zip
```

**Preconditions:**
1. 7.14.0 BC-4 Kibana cloud environment should be available.
2. Agent must be installed using Default policy having System.

**Steps to reproduce:**
1. Login to Kibana environment.
2. Navigate to Agent Logs tab.
3. Apply "error" filter log level dropdown.
4. Observe below errors:
```
11:30:19.283
elastic_agent
[elastic_agent][error] Elastic Agent status changed to: 'error'
11:30:19.290
elastic_agent
[elastic_agent][error] Failed to render configuration with latest context from composable controller: operator: failed to execute step sc-run, error: context canceled: context canceled
```

**Logs:**
[logs.zip](https://github.com/elastic/beats/files/6883091/logs.zip)

**Default Policy:**
[elastic-agent (1).zip](https://github.com/elastic/beats/files/6883089/elastic-agent.1.zip)

**Note:**
-  These logs are **not observed** on MAC, Linux tar, Linux .rpm, Linux .deb and Linux ARMx64 agents.
- We have reproduced these logs on two different Windows Machines.
- No impact on Agent is observed and these logs are generated only once after installation.

**Expected Result:**
"Failed to render configuration" error logs should not be available when windows agent is installed with only System integration.

**Screenshots:**
![2](https://user-images.githubusercontent.com/77374876/127104592-efc118d1-2585-4656-a54d-422a67baed16.JPG)

**Comment by amolnater-qasource** [comment]
@manishgupta-qasource Please review.

**Comment by manishgupta-qasource** [comment]
Reviewed & assigned to @EricDavisX

**Comment by EricDavisX** [comment]
@ruflin @michalpristas I'll ask you both, if you think this warrants any research for why Agent is throwing errors.  Ideally, Agent wouldn't, but as it happens only once and only on start up and we observe no other malfunctioning behaviors we could close it out.  Or leave it open for if someone wants to look at it as a tech-debt type item.

**Comment by ruflin** [comment]
It looks like this error is thrown in the transpiler of the dynamic inputs resolution: https://github.com/elastic/beats/blob/master/x-pack/elastic-agent/pkg/agent/application/pipeline/emitter/controller.go#L120 I suspect it is somehow related to the windows specific inputs in the system package or some provider which does not work as expected on windows. @blakerouse any ideas?

Suggestion to reproduce it: Use the same config as provided by @amolnater-qasource above in standalone mode on windows and check if the error happens also in standalone. If yes, remove input by input to see which inputs causes it.

**Comment by blakerouse** [comment]
`context cancelled` errors really should be caught and not logged. `context cancelled` are errors that are normal behavior in most situations. I think in this case that is logs from the fresh installation. The Elastic Agent is installed and started, then enrolled and then re-started to change to `managed` mode. That will cause the `context cancelled` error.

This is has no real impact other than ugly logs that are confusing. They should be cleaned up and removed.

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have observed new set of `error` logs on installing Windows 10 x64 agent on 7.14.0 BC-5 Kibana cloud environment.
- Windows 10 x64 agent is installed with Default policy having only system integration.

**Screenshot:**
![3](https://user-images.githubusercontent.com/77374876/127466395-c687e490-b6a1-44b3-869d-3d28c5572748.JPG)

**Logs:**
[Win-logs.zip](https://github.com/elastic/beats/files/6899406/Win-logs.zip)

cc: @ruflin @blakerouse 
Please let us know if anything else is required from our end.
Thanks
QAS

**Comment by EricDavisX** [comment]
that sounds like a new separate issue possibly, may be better off in its own ticket as such.

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We haven't observed these logs on 7.14.0 BC-6 as of now. We will be reporting new issue if we get any new `error` logs.

Thanks
